scripts/install-prettierx.js

```diff
@@ -4,33 +4,51 @@ const path = require("path");
 const shell = require("shelljs");
 const tempy = require("tempy");
 
+// [prettierx]: typescript/flow-parser optional dep support
+const pkg = require("../package.json");
+
 shell.config.fatal = true;
 
 const client = process.env.NPM_CLIENT || "yarn";
 
-module.exports = (packageDir) => {
-  const tmpDir = tempy.directory();
-  const file = shell.exec("npm pack", { cwd: packageDir }).stdout.trim();
-  shell.mv(path.join(packageDir, file), tmpDir);
-  const tarPath = path.join(tmpDir, file);
-
-  shell.exec(`${client} init -y`, { cwd: tmpDir, silent: true });
-  let installCommand = "";
+// [prettierx]: typescript/flow-parser optional dep support
+function getInstallCommand(args) {
   switch (client) {
     case "npm":
       // npm fails when engine requirement only with `--engine-strict`
-      installCommand = `npm install "${tarPath}" --engine-strict`;
-      break;
+      return `npm install ${args} --engine-strict`;
     case "pnpm":
       // Note: current pnpm can't work with `--engine-strict` and engineStrict setting in `.npmrc`
-      installCommand = `pnpm add "${tarPath}"`;
-      break;
+      return `pnpm add ${args}`;
     default:
       // yarn fails when engine requirement not compatible by default
-      installCommand = `yarn add "${tarPath}"`;
+      return `yarn add ${args}`;
   }
+}
 
-  shell.exec(installCommand, { cwd: tmpDir });
+module.exports = (packageDir) => {
+  const tmpDir = tempy.directory();
+  const file = shell.exec("npm pack", { cwd: packageDir }).stdout.trim();
+  shell.mv(path.join(packageDir, file), tmpDir);
+  const tarPath = path.join(tmpDir, file);
+
+  shell.exec(`${client} init -y`, { cwd: tmpDir, silent: true });
+
+  // [prettierx]: typescript/flow-parser optional dep support
+  shell.exec(getInstallCommand(`"${tarPath}"`), { cwd: tmpDir });
+
+  // [prettierx]: typescript/flow-parser optional dep support
+  const optionalDeps = new Set()
+
+  for (const dependency of Object.keys(pkg.peerDependenciesMeta)) {
+    if (pkg.peerDependenciesMeta[dependency].optional) {
+      optionalDeps.add(dependency);
+    }
+  }
+
+  if (optionalDeps.size > 0) {
+    shell.exec(getInstallCommand([...optionalDeps].join(" ")), { cwd: tmpDir });
+  }
```

Comment:
It would be ideal if we could find a way to do this with a little less code (code diff).

Reply:
I made a more concise set of changes without making the code too verbose or having to hard-code specific dependencies: https://github.com/brodybits/prettierx/pull/603/commits/dcdbd96b8dc58db287f29728a4dbfa3b3e22371d